### PR TITLE
10822: WordCloud 1.2.5: Stop word doesn't work if Word-breaking turne…

### PR DESCRIPTION
**Pull request by savvinsergey**
10822: WordCloud 1.2.5: Stop word doesn't work if Word-breaking turned off

### CODE REVIEW

File: src/visual.ts [hunk 1]
```diff
@@ -392,55 +392,89 @@ module powerbi.extensibility.visual {
...
-                        brokenStrings.push({
-                            text: splittedWord,
-                            textGroup: item.textGroup,
-                            count: item.count,
-                            index: item.index,
-                            selectionId: item.selectionId,
-                            color: item.color
-                        });
-                    });
+                    Array.prototype.push.apply(processedText, partOfProcessedText);
```
Comment (ignatvilesov) [inline src/visual.ts]: I'm not sure that using ```apply``` is a good way. The ```processedText``` requires three arguments, but this call applies only the first argument. I'd prefer to use explicit call of the function/method instead.

File: src/visual.ts [hunk 2]
```diff
@@ -392,55 +392,89 @@ module powerbi.extensibility.visual {
...
+            settings: WordCloudSettings) {
+
+            if (!settings.stopWords.show || !stopWords.length) {
+                return words;
+            }
+
+            return words.filter((value: string) => {
+                return value.length > 0 && !stopWords.some((removeWord: string) => {
+                        return value.toLocaleLowerCase() === removeWord.toLocaleLowerCase();
+                    });
```
Comment (ignatvilesov) [inline src/visual.ts]: The indent isn't correct. Please format this file using VS Code.

File: src/visual.ts [hunk 3]
```diff
@@ -392,55 +392,89 @@ module powerbi.extensibility.visual {
...
+            stopWords: string[],
+            settings: WordCloudSettings) {
+
+            if (!settings.stopWords.show || !stopWords.length) {
+                return words;
+            }
+
+            return words.filter((value: string) => {
+                return value.length > 0 && !stopWords.some((removeWord: string) => {
+                        return value.toLocaleLowerCase() === removeWord.toLocaleLowerCase();
```
Comment (ignatvilesov) [inline src/visual.ts]: The indent isn't correct. Please format this file using VS Code.

File: src/visual.ts (hunk 1 above)
Comment (savvinsergey) [inline src/visual.ts]: Done

File: src/visual.ts (hunk 3 above)
Comment (savvinsergey) [inline src/visual.ts]: Done

File: src/visual.ts (hunk 2 above)
Comment (savvinsergey) [inline src/visual.ts]: Done

File: src/visual.ts [hunk 4]
```diff
@@ -392,55 +392,89 @@ module powerbi.extensibility.visual {
...
 
-            if (!settings.general.isBrokenText) {
-                return words;
-            }
-
             words.forEach((item: WordCloudText) => {
                 if (typeof item.text === "string") {
                     let splittedWords: string[] = item.text
-                        .replace(punctuationRegExp, " ")
+                        .replace(punctuationRegExp, "")
```
Comment (ignatvilesov) [inline src/visual.ts]: Why do you remove all punctuation using ```punctuationRegExp``` as a pattern?
As far as I understand, this construction transforms ```Power,BI``` to ```PowerBI```. However, I suppose that this behavior is incorrect.

Comment (savvinsergey) [inline src/visual.ts]: Yes, you are right. Done

**Pull request merged**